### PR TITLE
allergies: Update json for new "input" policy

### PR DESCRIPTION
**Pull request by Stargator**
#996 Updated the schema for canonical data. So I am updating the allergies exercise's json to reflect the change.

### CODE REVIEW

File: exercises/allergies/canonical-data.json [hunk 1]
```diff
@@ -72,76 +78,109 @@
...
         },
         {
           "description": "allergic to just peanuts",
           "property": "list",
-          "score": 2,
-          "expected": ["peanuts"]
+          "input": {
+            "score": 2
+          },
+          "expected": [
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: Do the "expected"'s need to be changed or just the the addition of "input"?  I don't know that the present "expected" entries violate the schema.

Comment (Stargator) [reply exercises/allergies/canonical-data.json]: #996 Just mentions a change to `input`, I think `expected` is always one variable.

File: exercises/allergies/canonical-data.json [hunk 2]
```diff
@@ -15,7 +15,9 @@
         {
           "description": "no allergies means not allergic",
           "property": "allergicTo",
-          "score": 0,
+          "input": {
+            "score": 0
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: Does "input" really need "score"?  Why not just:
```text
"input": 0,
```
Of course like changes for other cases.  Unless we are really married to "score".  Others should perhaps weigh in on this point.

File: exercises/allergies/canonical-data.json [hunk 3]
```diff
@@ -34,7 +36,9 @@
         {
           "description": "is allergic to eggs",
           "property": "allergicTo",
-          "score": 1,
+          "input": {
+            "score": 1
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: "score" or no "score"?  See comment for first case.

File: exercises/allergies/canonical-data.json [hunk 4]
```diff
@@ -45,7 +49,9 @@
         {
           "description": "allergic to eggs in addition to other stuff",
           "property": "allergicTo",
-          "score": 5,
+          "input": {
+            "score": 5
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: "score" or no "score"?  See comment for first case.

File: exercises/allergies/canonical-data.json [hunk 5]
```diff
@@ -72,43 +78,57 @@
         {
           "description": "no allergies at all",
           "property": "list",
-          "score": 0,
+          "input": {
+            "score": 0
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: "score" or no "score"?  See comment for first case.

File: exercises/allergies/canonical-data.json [hunk 6]
```diff
@@ -72,43 +78,57 @@
...
+          },
           "expected": []
         },
         {
           "description": "allergic to just eggs",
           "property": "list",
-          "score": 1,
+          "input": {
+            "score": 1
```
Comment (rpottsoh) [inline exercises/allergies/canonical-data.json]: "score" or no "score"?  See comment for first case.   Ok, we get the idea!

File: exercises/allergies/canonical-data.json (hunk 2 above)
Comment (petertseng) [reply exercises/allergies/canonical-data.json]: don't care what y'all decide, but https://github.com/exercism/problem-specifications/issues/996#issuecomment-342740372 and https://github.com/exercism/problem-specifications/issues/996#issuecomment-342751835 are explicitly in favour of having `"input": {"score": 0}` not `"input": 0` therefore that is the only thing I support in https://github.com/exercism/problem-specifications/pull/998.

Comment (rpottsoh) [reply exercises/allergies/canonical-data.json]: I missed that detail.

**Pull request merged**